pkg/containers/datasource.go

```diff
@@ -0,0 +1,72 @@
+package containers
+
+import (
+	"encoding/json"
+
+	"github.com/aquasecurity/tracee/types/detect"
+)
+
+type SignaturesDataSource struct {
+	containers *Containers
+}
+
+func NewDataSource(c *Containers) *SignaturesDataSource {
+	return &SignaturesDataSource{
+		containers: c,
+	}
+}
+
+func (ctx SignaturesDataSource) Get(key interface{}) (map[string]interface{}, error) {
+	containerId, ok := key.(string)
+	if !ok {
+		return nil, detect.ErrKeyNotSupported
+	}
+	ctx.containers.mtx.RLock()
+	defer ctx.containers.mtx.RUnlock()
+	for _, cgroup := range ctx.containers.cgroupsMap {
```

Comment:
So every Get(container_id is O(n) - won't it be better to add another cgroupInfo map where the key is the container id and the value is a pointer to cgroup info so lookup will be O(1)?